### ISSUE TITLE
something not right with the model

### ISSUE THREAD

**Issue by johndpope** [johndpope/implicit-motion-function#23]
https://wandb.ai/snoozie/IMF/runs/wn2pxfep?nw=nwusersnoozie

![Screenshot from 2024-08-12 10-55-14](https://github.com/user-attachments/assets/9fe83e07-888f-4874-a410-fd4044935947)

to get to this working git commit

`git checkout -b "ethereal-monkey-1466" 91f23657c728a7c6fc7a4abb997273d36981ba6d
`

i have introduced a lot of flux into codebase
i reduce variables in config - by allowing changes to be turned on / off
```yaml

model:
  latent_dim: 32
  base_channels: 64
  num_layers: 4
  use_resnet_feature: False # new stuff Im playing around with
  use_mlgffn: False # new stuff Im playing around with
  use_enhanced_generator: False # new stuff Im playing around with
  use_skip: False# new stuff Im playing around with
```

the vanilla implementation mostly adheres to paper / supp. materials

https://github.com/hologerry/IMF

https://openaccess.thecvf.com/content/CVPR2024/papers/Gao_Implicit_Motion_Function_CVPR_2024_paper.pdf

https://openaccess.thecvf.com/content/CVPR2024/supplemental/Gao_Implicit_Motion_Function_CVPR_2024_supplemental.pdf

so why doesn't it work? 
I believe paper is wrong - on the one thing they needed to get right.
https://github.com/hologerry/IMF/issues/4

I adjust my model to use same dimensions for alignment - 
this works - but now there's some skewy with resnets / upconv / downconv.
this has a symptom of the model falling apart.

I believe there's something wrong with resblocks 

there maybe an error here?
![downconv](https://github.com/user-attachments/assets/732ec71e-cac4-43ed-a7fa-3b8bc7a4187c)
![featresblock](https://github.com/user-attachments/assets/7650277c-9637-47b7-8317-0275af27ae60)
![resblock](https://github.com/user-attachments/assets/760204e0-49ff-42f5-afe4-1b0d209cb4ba)
![upconv](https://github.com/user-attachments/assets/99870c6b-8663-42ca-944c-c3126a984ee4)

the code to exactly align to these - is commented out 
https://github.com/johndpope/IMF/blob/main/resblock.py#L15

![Screenshot from 2024-08-12 10-59-34](https://github.com/user-attachments/assets/448f3053-a138-4144-8a15-3b0fc6a11293)

when i switch these in - I get a first image - then the gradients stop flowing and training quits.
I don't know why....

I run into an issue where wandb had successfully trained but i hadn't commited code - so the run.sh auto commits - that way there's a safety commit to restore config. 

in the gradients.txt - there's exploding gradients all over - the training loops does clip gradients at 1.0 maybe need to drop this down to 0.5 

here on second video - model starts to fall apart.
![Screenshot from 2024-08-12 11-04-45](https://github.com/user-attachments/assets/d4a8e332-8e85-4f34-8b22-f688acf99b59)

Update - i drop clip gradients to 0.75 - 
something is being added...
https://wandb.ai/snoozie/IMF/runs/aged896s?nw=nwusersnoozie
2nd video
<img width="1315" alt="Screenshot 2024-08-12 at 1 51 03 pm" src="https://github.com/user-attachments/assets/df56bdaf-d37d-49ad-8756-8cbae186c3db">

git checkout -b "eternal-sun-1473" 24f9d4edff4aa8de17eb0eb9b0b19d1f1c57e79b

**Comment by johndpope** [comment]
idk - i fixed some major issues with modal training data contamination which was introducing blue/ red / black generated images. was side effect of dataset loading blank images.... 
i also introduce train_step that seems to be helping....
